src/zip_node.js

```diff
@@ -30,6 +45,42 @@ const zipNodeJs = async function ({ srcFiles, destPath, filename, mainFile, plug
   await endZip(archive, output)
 }
 
+// Zip a Node.js function file with esbuild
+const zipNodeJsWithEsbuild = async function ({
+  destFolder,
+  destPath,
+  externalModules = [],
+  filename,
+  mainFile,
+  pluginsModulesPath,
+}) {
+  const bundledFilePath = join(destFolder, filename)
+
+  await esbuild.build({
+    bundle: true,
+    entryPoints: [mainFile],
+    external: externalModules,
+    outfile: bundledFilePath,
+    platform: 'node',
+    target: ['es2017'],
+  })
+
+  const { archive, output } = startZip(destPath)
```

Comment:
Should those lines (from `startZip()` to `endZip()`) re-use the similar code above? The only difference is that there is only one `srcFile`, but we could use pass an array with a single element `[srcFile]` so that both functions use the same code. What do you think?

Reply:
I wanted to keep the two functions (`zipNodeJs` and `zipNodeJsWithEsbuild`) as compartmentalised as possible, at least during this initial phase. Not only because I expect the `zipNodeJsWithEsbuild` to change significantly over the coming weeks as we iron out the implementation, but also because I want to reduce as much as possible the possibility of us breaking the existing bundling behaviour.

I suggest we revisit this once the implementation matures a bit more. Do you think that's acceptable?

Reply:
Sound good :)